app/jobs/download_ead_job.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+##
+# Background job to download an EAD XML file from ArchiveSpace
+class DownloadEadJob < ApplicationJob
+  def perform(ead_id:, resource_uri:, repo_code:, data_dir: ENV.fetch('DATA_DIR', 'data'), index: false)
+    client = AspaceClient.new
+    ead_xml = client.resource_description(resource_uri)
+
+    file_name = ead_id.sub(/(\.xml)?$/i, '.xml')
+    directory = "#{data_dir}/#{repo_code}"
+
+    file_path = File.join(directory, file_name.to_s)
```

Comment:
I left this out because I wasn't sure it was a great idea for it to try to create this directory each of the several thousand times it might get called for a set of batched up jobs. What do you think about taking care of making sure the directory exists in some other method (to be written in a future PR) whose job it is to queue up large batches of download jobs?